### PR TITLE
Dockerfile for operator + Kube manifests

### LINKED ISSUE #1571: Operator manifests
Once #1570 is done, we should have some example operator manifests including:

* RBAC resources
* `Deployment`

### LINKED ISSUE #1576: Container resource requests and limits for Operator
The operator itself should have contain resources limits and requests configured in its `Deployment`.

### PR DESCRIPTION
**Pull request by tombentley**
### Type of change

- Enhancement / new feature

### Description

Adds a `Dockerfile` for building the operator, plus some YAMLs for standing up the operator in kubernetes.

I've made no attempt to integrate the operator image build with the rest of the build/CI in this PR. So it makes a start on #1570, but not enough to consider it fixed.

Fixes #1571
Fixes #1576

### CODE REVIEW

File: kroxylicious-operator/DEV_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+```bash
+minikube start --kubernetes-version=latest --driver=podman 
+````
+
+You should now be able to run the tests using `mvn`.
+
+If you want to run the `OperatorMain` (e.g. from your IDE, maybe for dubugging) then you'll need to install the CRD:
+
+```bash
+kubectl install -f src/main/resources/META-INF/fabric8
```
Comment (robobario) [inline kroxylicious-operator/DEV_GUIDE.md:18]: fails for me with:

```
kubectl install -f src/main/resources/META-INF/fabric8
error: unknown command "install" for "kubectl"
```

should this be `kubectl create`?

Comment (tombentley) [reply kroxylicious-operator/DEV_GUIDE.md:18]: `apply`

Comment (tombentley) [reply kroxylicious-operator/DEV_GUIDE.md:18]: `create` will also work, but `kubectl` will get shirty with you if you then try to use `apply` later

**Review by robobario — APPROVED** [review]
Apart from that one command it worked on my machine :+1:

**Pull request merged**